oceanmesh/signed_distance_function.py

```diff
@@ -54,10 +198,39 @@ def func(x):
         d, _ = tree.query(x, k=1)
         # d is signed negative if inside the
         # intersection of two areas and vice versa.
-        cond = numpy.logical_and(in_shoreline, in_boubox)
-        if flip:
-            cond = ~cond
+        cond = np.logical_and(in_shoreline, in_boubox)
         dist = (-1) ** (cond) * d
         return dist
 
     return Domain(shoreline.bbox, func)
+
+
+def multiscale_signed_distance_function(signed_distance_functions, verbose=True):
+    """Takes a list of :class:`signed_distance_function` objects and calculates a signed distance
+        function from each one.
+
+    Parameters
+    ----------
+    signed_distance_functions: a list of `signed_distance_function` objects
+
+    Returns
+    -------
+    union: a :class:`Union` object
+        The union of all signed distance functions
+    nests: a list of `Difference` objects
+        All inner domains are differenced from the domain above.
+
+    """
+    if verbose:
+        print("Building a multiscale signed distance function...")
+    assert isinstance(
+        signed_distance_functions, list
+    ), "`signed_distance_function` is not a list"
+    assert len(signed_distance_functions) > 1, "Use `signed_distance_function` instead"
+
+    nests = []
+    for ix1, sdf_base in enumerate(signed_distance_functions):
+        nests.append(Difference([sdf_base, *signed_distance_functions[ix1 + 1 :]]))
```

Comment:
I think the `Difference` operation should be performed with a slightly inflated/scaled interior `signed_distance_function` as the case the user employs a different shoreline in the nest than in the outer than, the `Difference` operation won't remove all the components of the coarse shoreline in the nest. 

However that requires an arbitrary scale, perhaps there's another way by taking another SDF primitive set op with the coarse domain's SDF...